Chun Yu FrontEndMentor Assignment (Rock Paper Scissors)

gunbux opened the pull request:
## Rock, Paper, Scissors, Lizard, Spock
Simple Rock Paper Scissors spinoff game made in React

### Comparison Images
`Original`
![desktop-step-1-bonus](https://user-images.githubusercontent.com/46131429/140074582-eb3fa5b1-55be-4224-92d6-a22383ed8137.jpg)
`Implementation`
![image](https://user-images.githubusercontent.com/46131429/140074657-cdd613e4-66d4-4d14-92de-41c8682ec6cd.png)
`Original`
![desktop-step-3-bonus](https://user-images.githubusercontent.com/46131429/140074718-e9a8e83a-a588-4205-812a-3e2dae607671.jpg)
`Implementation`
![image](https://user-images.githubusercontent.com/46131429/140074772-ab40561e-1e46-4c43-8509-3a9e324bbed5.png)
`Original`
![mobile-step-1-bonus](https://user-images.githubusercontent.com/46131429/140074869-42ce17e6-cbf3-49be-b06f-e031f19123a5.jpg)
`Implementation`
![image](https://user-images.githubusercontent.com/46131429/140074946-19bee7e2-2c9a-4b5e-9133-8a46be4bcacf.png)
`Original`
![mobile-step-4-bonus](https://user-images.githubusercontent.com/46131429/140074988-e71a4fab-f206-40fc-9570-9eb506b0b8ab.jpg)
`Implementation`
![image](https://user-images.githubusercontent.com/46131429/140075035-e60f1fb0-475f-45af-8758-2d35a5be249d.png)

---

gunbux committed:
Initial setup

gunbux committed:
Implemented UI and game states, refactored code

gunbux committed:
Added game logic

gunbux committed:
Added Lizard and Spock options, made UI Responsive

gunbux committed:
Minor changes

gunbux committed:
Merge branch 'main' of github.com:rhdevs/rhdevs-frontend-training-AY2122 into rock-paper-scissors

 Conflicts:
	src/pages/FrontendMentor/index.tsx
	src/routes/PATHS.ts
	src/routes/Routes.tsx

gunbux committed:
Major Code Refactoring

gunbux committed:
Minor changes in code from as per comments

xinyee20 commented on src/components/RockPaperScissors/PickContainer.tsx:
```diff
@@ -0,0 +1,23 @@
+  renderOption: (option: Option) => ReactElement
+}
+
+const PickContainer = ({ header, selection, renderOption }: Props) => {
```
consider just writing it as 'props: Props' ? so there won't be a long list of parameter

xinyee20 commented on src/components/RockPaperScissors/ResultState.tsx:
```diff
@@ -0,0 +1,48 @@
+  comSelection: string
+}
+
+const ResultState = ({ renderOption, result, handleReset, currentSelection, comSelection }: Props) => {
```
similarly, consider just writing it as 'props: Props' ? it's more accident proof too in cases like this where there is a higher chance a prop may be missed out

xinyee20 commented on src/components/RockPaperScissors/PickState.tsx:
```diff
@@ -0,0 +1,24 @@
+  renderOption: (option: Option) => ReactElement
+}
+
+const PickState = ({ currentSelection, comSelection, renderOption }: Props) => {
```
similarly, consider just writing it as 'props: Props' ?

xinyee20 commented on src/components/RockPaperScissors/SelectState.tsx:
```diff
@@ -0,0 +1,26 @@
+  optionList: Option[]
+}
+
+const SelectState = ({ renderOption, optionList }: Props) => (
```
similarly, consider just writing it as 'props: Props' ?

xinyee20 commented on src/components/RockPaperScissors/PickState.tsx:
```diff
@@ -0,0 +1,24 @@
+      <PickContainer renderOption={renderOption} header="THE HOUSE PICKED" selection={comSelection} />
+
+      {/*{renderPickContainer('YOU PICKED', currentSelection)}*/}
+      {/*{renderPickContainer('THE HOUSE PICKED', comSelection)}*/}
```
unused code can be removed

xinyee20 commented on src/components/RockPaperScissors/styles/Scoreboard.styled.ts:
```diff
@@ -0,0 +1,36 @@
+export const StyledScoreboard = styled.div`
+  width: 100%;
+  max-width: 800px;
+  //min-height: 150px;
```
delete if not used?

xinyee20 commented on src/components/RockPaperScissors/styles/Scoreboard.styled.ts:
```diff
@@ -0,0 +1,36 @@
+  width: ${(props) => props.imgSize ?? '2rem'};
+`
+export const StyledScoreboard = styled.div`
+  width: 100%;
```
can consider lowering it to maybe 85% and add margin so there is still some spacing between the scoreboard and the sides of the screen (will improve mobile view)

xinyee20 commented on src/components/RockPaperScissors/styles/Scoreboard.styled.ts:
```diff
@@ -0,0 +1,36 @@
+  max-width: 800px;
+  //min-height: 150px;
+  border-radius: 5px;
+  border: 2px solid #4e5f7d;
```
border can be increased slightly to maybe 3 and border-radius can be increase to maybe around 10px to look more like the original?

xinyee20 commented on src/pages/RockPaperScissors/styles/RockPaperScissors.styled.ts:
```diff
@@ -0,0 +1,42 @@
+export const MainContainer = styled.div`
+  width: 100%;
+  height: 100%;
+  min-width: 100vw;
```
I think this line is causing the horizontal scrolling, can consider removing it

xinyee20 reviewed (commented):
- Consider putting your file within FrontendMentor folder instead of under pages folder to improve organisation of the files
- Font has been provided for the score text and number, can be added in if you want it to look closer to the original
  font-weight: 600; (text) / 700; (number)
  font-family: 'Barlow Semi Condensed';
  color: hsl(229, 25%, 31%); (number)
*score number size can increase a bit more
- When the player won/lose/draw, the results doesn't come up if I don't click the screen. Maybe it will be better to show the results immediately without needing any action from the user?

gunbux committed:
Minor changes based on new comments.

gunbux commented:
> * Consider putting your file within FrontendMentor folder instead of under pages folder to improve organisation of the files
> * Font has been provided for the score text and number, can be added in if you want it to look closer to the original
>   font-weight: 600; (text) / 700; (number)
>   font-family: 'Barlow Semi Condensed';
>   color: hsl(229, 25%, 31%); (number)
>   *score number size can increase a bit more
> * When the player won/lose/draw, the results doesn't come up if I don't click the screen. Maybe it will be better to show the results immediately without needing any action from the user?

- Refactored code to be in FrontendMentor folder
- Changed Font to look more like the original (the method to import fonts in the notion wasn't working, used @import at index.css instead)
- increased score number size
- set timer to automatically switch to results after 1000ms

gunbux committed:
Updated font imports

limwenfeng commented:
LGTM!

gunbux committed:
Merge branch 'main' of github.com:rhdevs/rhdevs-frontend-training-AY2122 into rock-paper-scissors

 Conflicts:
	src/App.tsx
	src/pages/FrontendMentor/index.tsx
	src/routes/Routes.tsx
